examples/SerialRadio/SerialRadio.ino

```diff
@@ -109,13 +117,60 @@ void DisplayFrequency(RADIO_FREQ f)
 } // DisplayFrequency()
 
 
-/// Update the ServiceName text on the LCD display.
-void DisplayServiceName(char *name)
+/// Update the ServiceName text
+void DisplayRDSServiceName(char *name)
 {
-  Serial.print("RDS:");
+  Serial.print("RDS ServiceName:");
   Serial.println(name);
-} // DisplayServiceName()
+} // DisplayRDSServiceName()
 
+/// Update the text
+void DisplayRDSText(char *name)
+{
+  Serial.print("RDS Text:");
+  Serial.println(name);
+} // DisplayRDSText()
+
+/// Update the Date/Time
+void DisplayRDSTime(uint8_t hr, uint8_t mnt, uint32_t mjd, int8_t offset)
+{
+  Serial.print("  RDS Time: ");
+  if (hr < 10)
+    Serial.print('0');
+  Serial.print(hr);
+  Serial.print(':');
+  if (mnt < 10)
+    Serial.print('0');
+  Serial.print(mnt);
+  Serial.print("  Offset: ");
+  Serial.print(offset);
+
+  getMJDInfo(mjd, &mjd_info);
+  Serial.print("  Month: ");
+  Serial.print(mjd_info.month);
+  Serial.print("  Day: ");
+  Serial.print(mjd_info.day);
+  Serial.print("  Year: ");
+  Serial.println(mjd_info.year);
+} // DisplayRDSTime()
+
+
+void getMJDInfo(uint32_t mjd, MJD_INFO *mjdInfo)
+{
+  int _year = (int)((mjd - 15078.2) / 365.25);

```

Comment:
Please avoid floating-point arithmetic. I hope you find a pure integer based solution.